ShowerShow/Control/ScheduleController.cs

```diff
@@ -11,44 +11,121 @@
+        {
+            _logger.LogInformation("Getting schedules.");
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            HttpResponseData responseData = req.CreateResponse();
+            try
+            {
+                List<Schedule> schedules = await scheduleService.GetAllSchedules(UserId);
+                responseData.StatusCode = HttpStatusCode.OK;
 
+                return responseData;
 
-                return new OkObjectResult(schedule);
             }
             catch (Exception ex)
             {
                 // DEV ONLY
-                return new BadRequestObjectResult(ex.Message);
+                throw new Exception(ex.Message);
+            }
+        }
+        [Function("GetSchedule")]
+        [OpenApiOperation(operationId: "GetScheduleById", tags: new[] { "Schedules" })]
+        [OpenApiParameter(name: "ScheduleId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The Schedule ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Schedule), Description = "The OK response with the retrieved schedule")]
+        public async Task<Schedule> GetSchedule([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "schedule/{ScheduleId:Guid}")] HttpRequestData req, Guid ScheduleId)
+        {
+            _logger.LogInformation("Retrieving schedule.");
+
+            try
+            {
+                Schedule schedule = await scheduleService.GetScheduleById(ScheduleId);
+
+
+                return schedule;
+            }
+            catch (Exception ex)
+            {
+                // DEV ONLY
+                throw new Exception(ex.Message);
+            }
+        }
+        [Function("DeleteSchedule")]
+        [OpenApiOperation(operationId: "DeleteScheduleById", tags: new[] { "Schedules" })]
+        [OpenApiParameter(name: "ScheduleId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The Schedule ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Schedule), Description = "The OK response with the deleted schedule")]
+        public async Task<HttpResponseData> DeleteSchedule([HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route = "schedule/{ScheduleId:Guid}")] HttpRequestData req, Guid ScheduleId)
+        {
+            _logger.LogInformation("Deleting schedule.");
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            HttpResponseData responseData = req.CreateResponse();
+            try
+            {
+                Schedule schedule = null;
+
+                //this is to give priority to tasks
+                Task getId = Task.Run(() =>
```

Comment:
Everything works well, You just need two method from user end so you re dependent on me. You will need isAccountActive and ifUserExist methods. I will provide them to you once it is merged in dev. After that this branch is more than fine to be merged